messaging/src/main/java/org/axonframework/common/infra/ComponentDescriptor.java

```diff
@@ -16,24 +16,103 @@
 
 package org.axonframework.common.infra;
 
+import jakarta.annotation.Nonnull;
+
 import java.util.Collection;
 import java.util.Map;
 
+/**
+ * Contract towards describing the properties an (infrastructure) component might have.
+ * <p>
+ * The {@link ComponentDescriptor} implementation passed to a component by
+ * {@link DescribableComponent#describeTo(ComponentDescriptor)} determines both the structure and
+ * format of the output. This means the same component can be rendered in different formats (like JSON, XML, or a
+ * filesystem-like tree) depending on which {@link ComponentDescriptor} implementation is used. The component only needs
+ * to provide its properties and structure to the descriptor, which then handles the formatting details.
+ * <p>
+ * <strong>Handling Circular References:</strong> Component hierarchies may contain circular references where a
+ * {@link DescribableComponent} refers to another component that eventually references back to the original.
+ * {@link ComponentDescriptor} implementations must handle these circular dependencies to prevent infinite recursion.
+ *
+ * @author Allard Buijze
+ * @author Mitchell Herrijgers
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
 public interface ComponentDescriptor {
 
-    void describeProperty(String name, Object delegate);
+    /**
+     * Describe the given {@code object} with {@code this} descriptor for the given {@code name}.
+     * <p>
+     * If the {@code object} is a {@link DescribableComponent},
+     * {@link DescribableComponent#describeTo(ComponentDescriptor)} is invoked with {@code this} descriptor.
+     *
+     * @param name   The name for the {@code object} to describe.
+     * @param object The object to describe with {@code this} descriptor.
+     */
+    void describeProperty(@Nonnull String name, Object object);
 
-    void describeProperty(String name, Collection<?> delegate);
+    /**
+     * Describe the given {@code collection} with {@code this} descriptor for the given {@code name}.
+     * <p>
+     * The formatting of the {@code collection} typically takes a regular array structure.
```

Comment:
Nit: I think it is good to add here that if the objects in the `collection` are `DescribableComponents`, that those will be described as well.

Reply:
I am still wondering what you think of this nit of mine! So, perhaps more concrete: the implementation you have given checks if each entry is a `DescribableComponent`. If so, it acts is if `void describeProperty(@Nonnull String name, Object object)` is invoked, with the result pushed in an array. 

My suggestion is thus to state that "we check if the collection's items are describable components and, if such, we will invoke `DescribableComponent#describeTo` on them."

Reply:
I'm also still wondering why I have done it :P I'm on it.

Reply:
I hope now it's OK!